modules/nf-core/variancepartition/dream/templates/dream.R

```diff
@@ -0,0 +1,190 @@
+
+#' Parse out options from a string without recourse to optparse
+#'
+#' @param x Long-form argument list like --opt1 val1 --opt2 val2
+#'
+#' @return named list of options and values similar to optparse
+parse_args <- function(x) {
+    args_list <- unlist(strsplit(x, ' ?--')[[1]])[-1]
+    args_vals <- lapply(args_list, function(x) scan(text = x, what = 'character', quiet = TRUE))
+    args_vals <- lapply(args_vals, function(z) { length(z) <- 2; z })
+    parsed_args <- structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+    parsed_args[ !is.na(parsed_args) ]
+}
+
+#' Flexibly read CSV or TSV files
+#'
+#' @param file Input file
+#' @param header Passed to read.delim()
+#' @param row.names Passed to read.delim()
+#'
+#' @return output Data frame
+#'
+read_delim_flexible <- function(file, header = TRUE, row.names = NULL, check.names = TRUE, stringsAsFactors = FALSE) {
+    ext <- tools::file_ext(basename(file))
+    if (ext == "tsv" || ext == "txt") {
+        separator <- "\\t"
+    } else if (ext == "csv") {
+        separator <- ","
+    } else {
+        stop(paste("Unknown separator for", ext))
+    }
+    cat("Reading file", basename(file), "with", ext, "separator\n")
+    df <- read.delim( file, sep = separator, header = header, row.names = row.names,
+                    check.names = check.names, stringsAsFactors = stringsAsFactors)
+    return(df)
+}
+
+
+# Options list
+opt <- list(
+    output_prefix      = "$meta.contrast_id",       # Prefix for output files
+    count_file         = "$counts",                 # File containing raw counts
+    sample_file        = "$samplesheet",            # File containing sample information
+    contrast_variable  = "$meta.contrast_variable", # Variable for contrast (e.g., "treatment")
+    contrast_reference    = "$meta.contrast_reference",# Reference level for the contrast
+    contrast_target       = "$meta.contrast_target",   # Target level for the contrast (e.g., "mCherry")
+    blocking_factors   = "$meta.blocking_factors",  # Blocking variables (e.g., "sample_number")
+    sample_id_col      = "sample",    # Column name for sample IDs
+    threads            = "$task.cpus",              # Number of threads for multithreading
+    subset_to_contrast_samples = FALSE,            # Whether to subset to contrast samples
+    exclude_samples_col = NULL,                    # Column for excluding samples
+    exclude_samples_values = NULL,                 # Values for excluding samples
+    number             = Inf,                     # Maximum number of results
+    adjust.method      = "BH",                    # Adjustment method for topTable
+    p.value            = 1,                       # P-value threshold for topTable
+    lfc                = 0,                       # Log fold-change threshold for topTable
+    confint            = FALSE,                   # Whether to compute confidence intervals in topTable
```

Comment:
Is there a reason not to compute confidence intervals by default?
I probably wouldn't even have included this option - if someone doesn't like them they can subset the output dataframe.

Reply:
These default options were in the original `dev_tmp` module: 
https://github.com/nf-core/differentialabundance/blob/f2c9346d1b6e773e8164f67856e32834f9dfd9e6/modules/nf-core/dream/differential/resources/usr/bin/dream_de.R#L127-L131

I will remove them, then.